src/server/hook_func.c

```diff
@@ -5275,146 +5282,180 @@ sync_mom_hookfiles(void *minfo)
-						(pbs_errno != PBSE_MOM_REJECT_ROOT_SCRIPTS)) {
+			/* send with suffix if a regular hook file */
+			/* NOTE: There could be a hook named <resourcedef> */
+			/* and it's not the special <resourcedef> file */
+			/* that gets the special *RESCDEF aflags. In this */
+			/* case, the <resourcedef>.{HK,PY} would be sent */
+			/* instead */
+			/* send hook control file */
+			if (pact->action & MOM_HOOK_ACTION_SEND_ATTRS) {
+
+				snprintf(hookfile, MAXPATHLEN, "%s%s%s",
+					path_hooks, pact->hookname, HOOK_FILE_SUFFIX);
+				if ((phook->event & MOM_EVENTS) == 0) {
+					pact->action &= ~MOM_HOOK_ACTION_SEND_ATTRS;
+				} if ((PBSD_copyhookfile(conn, hookfile, 0, NULL) != 0) &&
+					(pbs_errno != PBSE_MOM_REJECT_ROOT_SCRIPTS)) {
+					snprintf(log_buffer, sizeof(log_buffer),
+						"errno %d: failed to copy hook file %s to %s:%d",
+						pbs_errno, hookfile,
+						minfo_array[i]->mi_host,
+						minfo_array[i]->mi_port);
+					log_event(PBSEVENT_DEBUG3,
+						PBS_EVENTCLASS_REQUEST, LOG_WARNING,
+						msg_daemonname, log_buffer);
+				} else {
+					if (pbs_errno != PBSE_MOM_REJECT_ROOT_SCRIPTS) {
 						snprintf(log_buffer, sizeof(log_buffer),
-							"errno %d: failed to copy hook file %s to %s:%d",
-							pbs_errno, hookfile,
-							minfo_array[i]->mi_host,
+							"successfully sent hook file %s to %s:%d",
+							hookfile, minfo_array[i]->mi_host,
 							minfo_array[i]->mi_port);
-						log_event(PBSEVENT_DEBUG3,
-							PBS_EVENTCLASS_REQUEST, LOG_WARNING,
-							"", log_buffer);
+						log_event(PBSEVENT_DEBUG,
+							PBS_EVENTCLASS_REQUEST, LOG_INFO,
+							msg_daemonname, log_buffer);
 					} else {
-						if (pbs_errno != PBSE_MOM_REJECT_ROOT_SCRIPTS) {
-							snprintf(log_buffer, sizeof(log_buffer),
-								"successfully sent hook file %s to %s:%d",
-								hookfile, minfo_array[i]->mi_host,
-								minfo_array[i]->mi_port);
-							log_event(PBSEVENT_DEBUG,
-								PBS_EVENTCLASS_REQUEST, LOG_INFO,
-								"", log_buffer);
-						} else {
-							snprintf(log_buffer, sizeof(log_buffer),
-								"warning: sending hook file %s to %s:%d got rejected (mom's reject_root_scripts=1)",
-								hookfile, minfo_array[i]->mi_host,
-								minfo_array[i]->mi_port);
-							log_event(PBSEVENT_DEBUG3,
-								PBS_EVENTCLASS_REQUEST, LOG_INFO,
-								"", log_buffer);
-						}
-						pact->action &= ~(MOM_HOOK_ACTION_SEND_ATTRS);
-						hook_track_save((mominfo_t *)minfo_array[i],
-							j);
-						action_done++;
+						snprintf(log_buffer, sizeof(log_buffer),
+							"warning: sending hook file %s to %s:%d got rejected (mom's reject_root_scripts=1)",
+							hookfile, minfo_array[i]->mi_host,
+							minfo_array[i]->mi_port);
+						log_event(PBSEVENT_DEBUG3,
+							PBS_EVENTCLASS_REQUEST, LOG_INFO,
+							msg_daemonname, log_buffer);
 					}
-				} /* send_attrs */
+					pact->action &= ~(MOM_HOOK_ACTION_SEND_ATTRS);
+					hook_track_save((mominfo_t *)minfo_array[i],
+						j);
```

Comment:
It's okay to be over the 80 line limit a little bit. AFAIK, we don't have a hard requirement for line width.